app/src/main/java/com/pps/globant/fittracker/mvp/model/SignUpModel.java

```diff
@@ -0,0 +1,77 @@
+package com.pps.globant.fittracker.mvp.model;
+
+import com.pps.globant.fittracker.mvp.model.DataBase.User;
+import com.pps.globant.fittracker.mvp.model.DataBase.UsersRepository;
+import com.squareup.otto.Bus;
+
+public class SignUpModel {
+    private Bus bus;
+    private User activeUser;
+    private UsersRepository usersRepository;
+
+    public SignUpModel(Bus bus, UsersRepository usersRepository) {
+        this.bus = bus;
+        this.usersRepository = usersRepository;
+    }
+
+    public User getUser() {
+        return activeUser;
+    }
+    //ROOM DATABASE
+
+    public void setUser(User user) {
+        activeUser = user;
+    }
+
+    public void getUserFromDbById() {
+        usersRepository.getById(activeUser.getId());
+    }
+
+    public void getUserFromDbById(int id) {
+        usersRepository.getById(id);
+    }
+
+    public void getUserFromDbBySocialNetworkId() {
+        usersRepository.getBySocialNetworkId(activeUser.getSocialNetworkId());
+    }
+
+    public void updateUserToDB() {
+        usersRepository.update(activeUser);
+    }
+
+    public void insertUserToDB() {
+        usersRepository.insert(activeUser);
+    }
+
+    public void deleteUserFromDb() {
+        usersRepository.delete(activeUser);
+    }
+
+    public boolean validateFirstName(String firstName) {
+        return (firstName.isEmpty() || firstName.length() < 3);
```

Comment:
add TextUtils.isEmpty(firstName) to avoid NPE

Reply:
done